lib/rbac/filterer.rb

```diff
@@ -252,12 +257,36 @@ def search(options = {})
               .includes(include_for_find).includes(exp_includes)
               .order(order)
 
-      scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
       scope = scope.limit(limit).offset(offset) if attrs[:apply_limit_in_sql]
-      targets = scope
 
-      unless options[:skip_counts]
-        auth_count = attrs[:apply_limit_in_sql] && limit ? targets.except(:offset, :limit, :order).count(:all) : targets.length
+      # putting this into an inner view will allow us to apply the virtual attributes to LIMIT rows not all of them.
+      if !options[:use_sql_view]
+        scope = include_references(scope, klass, include_for_find, exp_includes, skip_references)
+        targets = scope
+
+        unless options[:skip_counts]
+          auth_count = attrs[:apply_limit_in_sql] && limit ? targets.except(:offset, :limit, :order).count(:all) : targets.length
+        end
+      else
+        if !options[:skip_counts] && (attrs[:apply_limit_in_sql] && limit)
+          auth_count = scope.except(:offset, :limit, :order).count(:all)
+        end
+        if attrs[:apply_limit_in_sql] && limit && extra_cols
+          scope.includes_values = []
+          scope.references_values = []
+          scope = add_joins(klass, scope, include_for_find)
```

Comment:
Aside what we have talked about out of band about seeing if we can fore-go `add_joins` for now, can `scope` in the context of this block be a separate variable?

Seems like we be creating a different scope for what results in `Arel.sql("(#{scope.to_sql})"`, and only apply it to the `scope` var when doing `.from`.

Possibly `nested_from = add_joins(...)` ?

Reply:
got in the nested_from gook.
had to remove the `references` and go with the joins - it was causing all sorts of issues.

frustrated I'm dealing with decoding the joins clauses. But keeping the requires/includes in the inline view (`from()` part) was throwing random errors due to where clauses not finding references.